### ISSUE TITLE
Create Helm Chart for K8s Deployment

### ISSUE THREAD

**Issue by uvulpos** [Clivern/Lynx#59]
## Is your feature request related to a problem? Please describe.
As a DevOps Engineer, I have a given base Infrastructure on hand where I want to deploy this software to deploy another cluster with terraform in the cloud.

## Describe the solution you'd like
...so I need to deploy it inside my existing cluster via helm. So, I don't want to define my own Kubernetes resources, so I would appreciate it, if we could share one resource for this inside the community

**Comment by Clivern** [comment]
Hi @uvulpos, That makes sense indeed. My knowledge with Helm is limited so i need to check if is it better to have PostgresSQL as dependency to run inside the cluster? or shall i let the user to decide whether to use PostgreSQL from outside the cluster. Honestly i would feel much better having PostgresSQL outside the cluster.

**Comment by uvulpos** [comment]
Hi @Clivern ,
general speaking is both possible, but it's rather common to just release your helm configuration for your applications and dependencies as postgres is managed elsewhere e.g. with an kubernetes operator. So you would need a kubernetes deployment and a service definition and place helm variables in it

**Comment by uvulpos** [comment]
I can do a basic pr to create the structure. We can finalize it later, but let's do an MVP first

**Comment by Clivern** [comment]
Yes, please feel free to open a PR.

**Comment by ankek** [comment]
Hi there, some updates from my side. K8S is not a good place to store data.  There are a lot of example with data loss. My suggestion to use external (outside of K8S) Database and store data there to avoid data loss for Production-ready environments.

**Comment by uvulpos** [comment]
@ankek I would agree, speaking about deploying a own database. But this is not the case and not the wanted solution. Of you already deployed an application to k3s / k8s, you most likely deployed an application with some sort of need for storage. Then you either deploy a database as stateful set on a specific node, or, if you really need to scale, there are hosted databases as elastic postgres in azure (as example). And we only ship the application. No database. So the user is responsible, how to store that kind of data

**Comment by ankek** [comment]
@uvulpos  I do not argue with you. For test cases/ scenarios of course we would need DB in helm deployment. I meant that it would be great to let the people make decisions about the DB backend and its location (inside or outside K8S).
And if we are talking about the Helm chart it makes sense to have this option from the very beginning.

**Comment by uvulpos** [comment]
You can just start the DB from docker compose file in root and connect to this for testing
```
docker compose up db
```

**Comment by uvulpos** [comment]
Of course, you can define the connection string as a value in helm. There is the concept of variables. But including a database is still silly for our use case and would just mean lots of extra work.

How I mentioned: you can do a simple stateful set deployment with a single node (for playing around) and put it inside the helm chart, but we can't scale anyway and so many important features as backups or different storage providers would be missing or have no support. Also, you would waste resources if every single application would bring their own database. Lynx will never be the only application inside the cluster. It would just add tickets from unsatisfied users

Victor from DOT also explained it very well:

https://youtu.be/Ny9RxM6H6Hg?si=pgNV9cjHznlEKhav

**Comment by uvulpos** [comment]
When you want to deploy a database at the same time, I would recommend helmfiles

https://github.com/helmfile/helmfile

**Comment by aron-muon** [comment]
Hello, we at Muon Space have developed a fork which includes a released helm chart (where postgres is counted as an external dependency and isn't deployed within the chart). Feel free to check it out.
https://github.com/Muon-Space/Lynx
